server/src/auth/strategies.ts

```diff
@@ -186,13 +186,28 @@ export function validateAndCacheHostName(request: Request, response: Response, n
     }
 }
 
-export function createLink(request: Request, link: string): string {
+// export function createLink(request: Request, link: string): string {
+//     if (link[0] === "/") {
+//         link = link.substring(1);
+//     }
+//     if ((request.secure && getExternalPort(request) === 443) || (!request.secure && getExternalPort(request) === 80)) {
+//         return `http${request.secure ? "s" : ""}://${request.hostname}/${link}`;
+//     } else {
+//         return `http${request.secure ? "s" : ""}://${request.hostname}:${getExternalPort(request)}/${link}`;
+//     }
+// }
+
+export function createLink(request: Request, link: string, proto?: string): string {
+    if (!proto) {
+        proto = "http";
+    }
     if (link[0] === "/") {
         link = link.substring(1);
     }
-    if ((request.secure && getExternalPort(request) === 443) || (!request.secure && getExternalPort(request) === 80)) {
-        return `http${request.secure ? "s" : ""}://${request.hostname}/${link}`;
+    if ((request.secure && getExternalPort(request) === 443)
```

Comment:
what do 443 and 80 stand for? Also I feel ternaries make this feel more complicated than it is

Reply:
This is checkin2 code that I stole since it uses subscriptions, 80 and 443 and HTTP ports